### DISCUSSION TITLE
Support for Selecting Specific Paths from JSON Objects in Evaluation Prompts

### DISCUSSION THREAD

**Discussion by divyanshubhuta** [langfuse/langfuse#3237] [Ideas]
### Describe the feature or potential improvement

Currently, the evaluation prompts only support using the entire metadata field from a JSON object, without the ability to select specific paths within the object. It would be beneficial to enhance this functionality by allowing users to specify and extract particular paths or keys from the JSON object for more targeted evaluation. This feature would improve precision and flexibility when working with complex metadata structures.

**Comment by marcklingen** [comment]
thanks for sharing, this is also interesting to extract from structured input/output fields as well

**Comment by marcklingen** [comment]
Additional context on this in #3352 by @stepanogil

> ### Describe the feature or potential improvement
> 
> 
> This is how you currently map variables in the model-based Evaluation (Beta) module:
> 
> ![image](https://github.com/user-attachments/assets/6b4bb862-279a-4b04-befe-27dde85adb8c)
> 
> i.e. you map whole returns of Input or Output or Metadata to a variable e.g. {{query}}
> 
> This is an example of how I populate the a Trace Input:
> 
> ![image](https://github.com/user-attachments/assets/2731ad93-d703-431d-b958-c06a65891b1f)
> 
> This means when I do Evals, this whole json blob is passed on as the value of my {{query}}:
> 
> ![image](https://github.com/user-attachments/assets/836965f6-41ef-4eee-8149-d0a7a2a20eb9)
> 
> Which I don't want to do as all the metadata will just introduce a lot of noise to the eval.
> What I want to do is map my {{query}} to particular kv pair inside of the Input. Example below:
> 
> ![image](https://github.com/user-attachments/assets/f2727a46-b91b-4f53-9122-6bb9a5350a75)
> 
> So I want another dropdown here where I can configure to which key I want to map the {{query}} variable.
> So I can input something like: `Input["conversation_history"][0]["content"]` or something to this effect.
> 
> ![image](https://github.com/user-attachments/assets/62256c3b-e7e6-4e91-b913-9b3cf8121225)

**Comment by maxdeichmann** [comment]
@stepanogil and @divyanshubhuta we built and released this yesterday. Please let us know if this works for you.

https://langfuse.com/changelog/2025-02-06-llm-as-a-judge-jsonpath

**Reply by stepanogil** [reply]
awesome! 🚀
